components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/internal/IdentitySAMLSSOServiceComponent.java

```diff
@@ -409,4 +409,27 @@ protected void unsetExtensionProcessor(SAMLExtensionProcessor extensionProcessor
         SAMLSSOUtil.removeExtensionProcessors(extensionProcessor);
     }
 
+    /**
+     * Add dependency to the ApplicationManagementService.
+     */
+    @Reference(
+            name = "identity.application.management.service",
+            service = ApplicationManagementService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetApplicationManagementService"
+    )
+    protected void setApplicationManagementService(ApplicationManagementService applicationManagementService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("ApplicationManagementService is available");
+        }
+    }
+
+    protected void unsetApplicationManagementService(ApplicationManagementService applicationManagementService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("ApplicationManagementService is not available");
```

Comment:
Is this debug log correct?

Reply:
This module depends on the activate method of the application management service. Therefore in the context of this component, the unset of the application management service means it is not available in the OSGi env. WDYT?

Reply:
For this context, debug log in the set method is appropriate. 

Unset will be called when the said service is uninstalled from the OSGi runtime. So regardless of the context, it will be saying that the service will no longer be available in the OSGi runtime.

Reply:
Fixed with e03d85290615f6fc225a0e674a0b00d83f5d3c19.